models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -430,10 +435,8 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
                     fclose($stream);
                 }
 
-                if ($statusCacheEnabled) {
-                    if ($imageInfo = @getimagesize($tmpFsPath)) {
-                        $asset->getDao()->addToThumbnailCache($config->getName(), $filename, filesize($tmpFsPath), $imageInfo[0], $imageInfo[1]);
-                    }
+                if ($statusCacheEnabled && $asset instanceof Asset\Image) {
+                    $asset->getThumbnail($config->getName())->readDimensionsFromFile(); //update thumbnail dimensions to cache
```

Comment:
I'm wondering if this is really a good idea, since this is calling the whole generation again, although we have already a local file at this point. 
Maybe it's better if we do something like that on the `Asset` class: 

```php
/**
     * @internal
     */
    public function addThumbnailFileToCache(string $localFile, string $filename, ?Config $config): void 
    {
        if ($imageInfo = @getimagesize($localFile)) {
            $dimensions = [
                'width' => $imageInfo[0],
                'height' => $imageInfo[1],
            ];
            
            if ($config) {
                $this->getDao()->addToThumbnailCache(
                    $config->getName(),
                    $filename,
                    filesize($localFile),
                    $dimensions['width'],
                    $dimensions['height']
                );
            }
        }
    }
```

it's then generic, we can use it in `readDimensionsFromFile()` and in `Processor` and we don't need to make some extra miles, WDYT?

Reply:
yes, agree that we don't need to re-generate the local file on `process`. I refactored it as per the suggestion.